.github/workflows/macos.yaml

```diff
@@ -30,7 +30,7 @@ on:
 
 jobs:
   ci:
-    runs-on: macos-latest
+    runs-on: macos-13
```

Comment:
This is temporary!

See: https://github.com/actions/setup-python/issues/547

In particular, the architecture on `macos-latest` (`macos-14`) machines means that mixed-architecture dylibs are produced by `gcc`.

See for [example][1] a failed build:

```
...
Successfully built bezier
nox > delocate-wheel --wheel-dir /var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/tmpraciqawj --verbose /var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/tmpizrl0uv3/bezier-2023.7.29.dev1-cp311-cp311-macosx_10_9_universal2.whl
Fixing: /var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/tmpizrl0uv3/bezier-2023.7.29.dev1-cp311-cp311-macosx_10_9_universal2.whl
Traceback (most recent call last):
  File "/Users/runner/work/bezier/bezier/.nox/doctest/bin/delocate-wheel", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/runner/work/bezier/bezier/.nox/doctest/lib/python3.11/site-packages/delocate/cmd/delocate_wheel.py", line 110, in main
    copied = delocate_wheel(
             ^^^^^^^^^^^^^^^
  File "/Users/runner/work/bezier/bezier/.nox/doctest/lib/python3.11/site-packages/delocate/delocating.py", line 971, in delocate_wheel
    copied_libs = delocate_path(
                  ^^^^^^^^^^^^^^
  File "/Users/runner/work/bezier/bezier/.nox/doctest/lib/python3.11/site-packages/delocate/delocating.py", line 509, in delocate_path
    lib_dict = tree_libs_from_directory(
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/runner/work/bezier/bezier/.nox/doctest/lib/python3.11/site-packages/delocate/libsana.py", line 377, in tree_libs_from_directory
    return _tree_libs_from_libraries(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/runner/work/bezier/bezier/.nox/doctest/lib/python3.11/site-packages/delocate/libsana.py", line 304, in _tree_libs_from_libraries
    for depending_path, install_name in get_dependencies(
  File "/Users/runner/work/bezier/bezier/.nox/doctest/lib/python3.11/site-packages/delocate/libsana.py", line 102, in get_dependencies
    for install_name in get_install_names(lib_fname):
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/runner/work/bezier/bezier/.nox/doctest/lib/python3.11/site-packages/delocate/tools.py", line 559, in get_install_names
    names_data = _check_ignore_archs(_parse_otool_install_names(otool.stdout))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/runner/work/bezier/bezier/.nox/doctest/lib/python3.11/site-packages/delocate/tools.py", line 413, in _check_ignore_archs
    raise NotImplementedError(
NotImplementedError: This function does not support separate values per-architecture: {'x86_64': [('/usr/lib/libSystem.B.dylib', '1.0.0', '1336.0.0')], 'arm64': [('@rpath/libbezier.2023.dylib', '2023.0.0', '2023.7.28'), ('/usr/lib/libSystem.B.dylib', '1.0.0', '1336.0.0')]}
nox > Command delocate-wheel --wheel-dir /var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/tmpraciqawj --verbose /var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/tmpizrl0uv3/bezier-2023.7.29.dev1-cp311-cp311-macosx_10_9_universal2.whl failed with exit code 1
...
```

[1]: https://github.com/dhermes/bezier/actions/runs/9554636526/job/26336137121